src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionResolveHandler.cs

```diff
@@ -0,0 +1,279 @@
+            var runAsCommand = false;
+
+            var applyChangesOperations = operations.Where(operation => operation is ApplyChangesOperation);
+            if (applyChangesOperations.Any())
+            {
+                var workspaceEdits = await ComputeWorkspaceEdits(applyChangesOperations, document!, cancellationToken).ConfigureAwait(false);
+                if (workspaceEdits.Any())
+                {
+                    codeAction.Edit = new LSP.WorkspaceEdit { DocumentChanges = workspaceEdits };
+                }
+                else
+                {
+                    // The workspace edit is something we don't currently support, like adding a new document.
+                    runAsCommand = true;
+                }
+            }
+
+            // Set up to run as command on the server instead of using WorkspaceEdits.
+            var commandOperations = operations.All(operation => !(operation is ApplyChangesOperation));
+            if (commandOperations || runAsCommand)
+            {
+                codeAction.Command = SetCommand(codeAction, data);
+            }
+
+            return codeAction;
+
+            // Local functions
+            static async Task<TextDocumentEdit[]> ComputeWorkspaceEdits(
+                IEnumerable<CodeActionOperation> applyChangesOperations,
+                Document document,
+                CancellationToken cancellationToken)
+            {
+                using var _ = ArrayBuilder<TextDocumentEdit>.GetInstance(out var textDocumentEdits);
+                foreach (ApplyChangesOperation applyChangesOperation in applyChangesOperations)
+                {
+                    var solution = document.Project.Solution;
+                    var changes = applyChangesOperation.ChangedSolution.GetChanges(solution);
+                    var projectChanges = changes.GetProjectChanges();
+
+                    // TO-DO: If the change involves adding a document, execute via command instead of WorkspaceEdit
+                    // until adding documents is supported in LSP: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1147293/
+                    // After support is added, remove the below if-statement and add code to support adding documents.
+                    var addedDocuments = projectChanges.SelectMany(
+                        pc => pc.GetAddedDocuments().Concat(pc.GetAddedAdditionalDocuments().Concat(pc.GetAddedAnalyzerConfigDocuments())));
+                    if (addedDocuments.Any())
+                    {
+                        return textDocumentEdits.ToArray();
+                    }
+
+                    var changedDocuments = projectChanges.SelectMany(pc => pc.GetChangedDocuments());
+                    var changedAnalyzerConfigDocuments = projectChanges.SelectMany(pc => pc.GetChangedAnalyzerConfigDocuments());
+                    var changedAdditionalDocuments = projectChanges.SelectMany(pc => pc.GetChangedAdditionalDocuments());
```

Comment:
@CyrusNajmabadi Is there a way to specify get all diffs between two solutions, including analyzer and additional documents?  Rather than having to go document by document?

Reply:
not that im' aware of.

Reply:
also, we are specially handling adding a doc, but not removing?  the approach here is very centered about supporting our code actions.  but would nto necessarily work for other code actions (or if we add code actinos in the future that do more than add-docs or edit docs.

Reply:
my concern still applies here.  i'm not sure what the philosophy is, but it seems like we're not handling the full set of workspace edits that would normally be possible.  that makes me uneasy and makes me feel like we could easily ship broken stuff.  i.e. someone writes a feature like "move type" and expects to be able to remove a file.  We write it and test it and validate it in VS.  but then in LSP itis just broken and we wouldn't know until someone hits it.

i do not like things like that because it just means different parties have to know all the implemented/unimplemnted capabilities of all these subcomponents.

Reply:
I added code so that any code action that executes as a command should (for now) be executed as a command.